spec/support/download_helper.rb

```diff
@@ -42,6 +42,11 @@ def self.remove_downloads
 
   def self.create_directory
     PATH.parent.mkdir unless PATH.parent.exist?
-    PATH.mkdir
+    PATH.mkdir unless PATH.exist?
+  rescue Errno::EEXIST
+    # Saw this on travis, suggesting a race condition
+    # probably chrome creating the directory for the actual download.
+    # We retry, as we could potentially have failed creating the parent directory.
+    retry
```

Comment:
So is this solving an issue where the directory does not exist at line 8, but by the time it's got to line 45 it does exist? 
When would Errno:EEXIST get thrown?

Thanks

Reply:
Yep, Errno:EEXIST gets thrown by PATH.mkdir (or PATH.parent.mkdir) if the directory already exists. Its actually even possible to get the exception thrown on line 44, as the line isn't atomic, so you could check for the existence, and find it missing, have chrome create the directory immediately after, and then try to create the directory yourself.

I've only seen the error the once, so it probably isn't a common occurrence, but I hate seemingly random test failures.